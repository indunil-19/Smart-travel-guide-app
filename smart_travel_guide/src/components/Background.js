import React from "react";
import {
  ImageBackground,
  StyleSheet,
  KeyboardAvoidingView,
} from "react-native";

const Background = (props) => (
  <ImageBackground
    source={props.backgroundImage}
    resizeMode={props.resizeMode}
    style={{ ...styles.background, ...props.style }}
    {...props}
  >
    <KeyboardAvoidingView
      style={{ ...styles.container, ...props.style }}
      behavior="padding"
    >
      {props.children}
    </KeyboardAvoidingView>
  </ImageBackground>
);

Background.defaultProps = {
  // backgroundImage: require("../assets/background_dot.png"),
  // resizeMode: "repeat",
};

const styles = StyleSheet.create({
  background: {
    flex: 1,
    width: "100%",
    backgroundColor: "#FFFFFF",
  },
  container: {
    flexGrow: 1,
    // padding: 20,
    width: "100%",
    // maxWidth: 340,
    alignSelf: "center",
    alignItems: "center",
    justifyContent: "space-evenly",
  },
});

export default Background;
