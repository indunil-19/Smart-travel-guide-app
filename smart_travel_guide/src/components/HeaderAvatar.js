import React, { useContext } from "react";
import { Avatar } from "react-native-paper";
import { TouchableOpacity } from "react-native";
import { useNavigation } from "@react-navigation/native";

import { AppContext } from "../context/AppContext";

import { theme } from "../core/theme";

export const HeaderAvatar = () => {
  const { state } = useContext(AppContext);
  const navigation = useNavigation();
  if (state.pic) {
    return (
      <TouchableOpacity onPress={() => navigation.navigate("Profile")}>
        <Avatar.Image
          source={{ uri: state.pic }}
          size={30}
          onPress={() => navigation.navigate("Profile")}
        />
      </TouchableOpacity>
    );
  } else {
    return (
      <TouchableOpacity onPress={() => navigation.navigate("Profile")}>
        <Avatar.Icon
          icon="account"
          color={theme.colors.primary}
          size={30}
          style={{ backgroundColor: "white" }}
        />
      </TouchableOpacity>
    );
  }
};
