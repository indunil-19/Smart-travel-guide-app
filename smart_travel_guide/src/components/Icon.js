export const icon = (val) => {
  var answer = "";
  switch (val) {
    case "Dry":
      answer = "weather-sunny";
      console.log(val);
      break;
    case "Intermediate":
      answer = "snowflake";
      break;
    case "Wet":
      answer = "weather-rainy";
      break;
    case "Hiking":
      answer = "hiking";
      break;
    case "Camping":
      answer = "fire";
      break;
    case "Swimming":
      answer = "swim";
      break;
    case "Reliogous":
      answer = "church";
      break;
    case "Catholic":
      answer = "church";
      break;
    case "Buddhism":
      answer = "buddhism";
      break;
    case "Hindu":
      answer = "hinduism";
      break;
    case "Islam":
      answer = "islam";
      break;
    case "Beaches":
      answer = "beach";
      break;
    case "Gardens":
      answer = "flower";
      break;
    case "Natural":
      answer = "tree-outline";
      break;
    case "Animal":
      answer = "elephant";
      break;
    case "Ancient":
      answer = "castle";
      break;
    case "Parks":
      answer = "dog-side";
      break;
    case "Riding boats":
      answer = "rowing";
      break;
    case "Surfing":
      answer = "weather-windy";
      break;

    default:
      answer = "star-outline";
  }
  return answer;
};
