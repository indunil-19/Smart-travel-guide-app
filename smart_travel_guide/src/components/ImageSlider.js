import React, { useState, useEffect } from "react";
import { View, Text, StyleSheet, Image } from "react-native";
import { Dimensions } from "react-native";
import Carousel from "react-native-snap-carousel";
import { scrollInterpolator, animatedStyles } from "../core/animation";
import { Card } from "react-native-paper";
import { Button } from "react-native-paper";
const SLIDER_WIDTH = Dimensions.get("window").width;
const ITEM_WIDTH = Math.round(SLIDER_WIDTH * 0.7);
const ITEM_HEIGHT = Math.round((ITEM_WIDTH * 3) / 4);

const ImageSlider = (props) => {
  const [imgRef, setImgRef] = useState([]);

  const [refLoading, setRefLoading] = useState(true);
  useEffect(() => {
    setImgRef((cur) => props.photos);
    setRefLoading(false);
  }, [props.photos]);

  const render = ({ item, index }) => {
    return (
      <>
        {!refLoading && (
          <View style={styles.itemContainer}>
            <Image
              fadeDuration={2000} //* image fade in duration on 1st load after that image is cached in memory used in networkloaded images can be also used in loacl images
              source={{
                uri: `https://maps.googleapis.com/maps/api/place/photo?maxwidth=800&photo_reference=${imgRef[index]}&key=AIzaSyCB9FiwGVeEmdfBAwxiQpPuz0fsDMiwPWY`,
              }}
              style={styles.image}
              resizeMode="cover"
            />
          </View>
        )}
      </>
    );
  };
  return (
    <>
      {!refLoading && (
        <View style={{ width: "100%" }}>
          <Carousel
            data={props.photos}
            layout={"stack"}
            renderItem={render}
            sliderWidth={SLIDER_WIDTH}
            itemWidth={ITEM_WIDTH}
            containerCustomStyle={styles.carouselContainer}
            inactiveSlideShift={0}
            scrollInterpolator={scrollInterpolator}
            slideInterpolatedStyle={animatedStyles}
            useScrollView={true}
          />
        </View>
      )}
    </>
  );
};
const styles = StyleSheet.create({
  card: {
    borderRadius: 5,
    width: 250,
  },
  carouselContainer: {
    marginVertical: 10,
  },
  itemContainer: {
    width: ITEM_WIDTH,
    height: ITEM_HEIGHT,
  },
  image: {
    width: " 100%",
    height: "100%",
    borderRadius: 5,
  },
  itemLabel: {
    color: "white",
    fontSize: 24,
  },
  counter: {
    marginTop: 25,
    fontSize: 30,
    fontWeight: "bold",
    textAlign: "center",
  },
});

export default ImageSlider;
