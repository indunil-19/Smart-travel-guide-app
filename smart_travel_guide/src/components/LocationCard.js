import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { Avatar, Button, Title, Card, Paragraph } from "react-native-paper";

const LocationCard = (props) => {
  // const [link, setLink] = useState("");
  // useEffect(() => {
  //   setLink(
  //     `https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photo_reference=${photo}&key=AIzaSyCB9FiwGVeEmdfBAwxiQpPuz0fsDMiwPWY`
  //   );
  // }, [photo]);
  return (
    <Card>
      <Card.Title title={props.name} />
      {console.log(props)}
    </Card>
  );
};

const styles = StyleSheet.create({});
export default LocationCard;
