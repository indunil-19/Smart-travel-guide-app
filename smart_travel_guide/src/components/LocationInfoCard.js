import React, { useState, useEffect } from "react";
import { StyleSheet } from "react-native";
import { Card, IconButton, Button } from "react-native-paper";
import { AirbnbRating } from "react-native-ratings";

import { theme } from "../core/theme";

export const LocationInfoCard = (props) => {
  const [imgLink, setImgLink] = useState("");
  const [isLoading, setLoaded] = useState(true);

  useEffect(() => {
    setImgLink(
      `https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photo_reference=${props.location.photos[0].photo_reference}&key=AIzaSyCB9FiwGVeEmdfBAwxiQpPuz0fsDMiwPWY`
    );
    setLoaded(false);
  }, [props.location.photos[0].photo_reference]);
  return (
    <Card elevation={6} style={styles.card}>
      {!isLoading && (
        <Card.Cover source={{ uri: imgLink }} style={styles.cover} />
      )}
      {isLoading && (
        <Card.Cover
          source={require("../assets/img/load-loading.gif")}
          style={styles.cover}
        />
      )}

      <Card.Title
        title={props.location.name}
        subtitle={props.location.formatted_address}
        right={() => (
          <IconButton
            icon={{ uri: props.location.icon }}
            color={theme.colors.primary}
            size={15}
          />
        )}
        style={{ paddingHorizontal: 10 }}
      />
      <Card.Content
        style={{
          flexDirection: "row",
          justifyContent: "space-between",
          paddingRight: 5,
        }}
      >
        <AirbnbRating
          count={5}
          defaultRating={props.location.rating}
          selectedColor={theme.colors.primary}
          showRating={false}
          size={20}
          isDisabled
        />
        <Button>Learn More</Button>
      </Card.Content>
    </Card>
  );
};

const styles = StyleSheet.create({
  card: {
    width: "90%",
    alignSelf: "center",
    marginVertical: 12,
    borderRadius: 10,
  },
  cover: {},
});
