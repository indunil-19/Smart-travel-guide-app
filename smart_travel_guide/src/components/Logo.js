import React from "react";
import { Image, StyleSheet } from "react-native";

const Logo = () => (
  <Image source={require("../assets/img/logo.png")} style={styles.image} />
);

const styles = StyleSheet.create({
  image: {
    marginBottom: 20,
  },
});

export default Logo;
