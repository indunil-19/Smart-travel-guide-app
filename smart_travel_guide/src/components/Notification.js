import React, { useState, useEffect, useContext } from "react";
import { Snackbar } from "react-native-paper";
import { AppContext } from "../context/AppContext";

export const Notification = () => {
  const [isVisible, setVisible] = useState(false);
  const [notification, setNotification] = useState({});
  const { state, dispatch } = useContext(AppContext);

  useEffect(() => {
    if (state.notification) {
      setVisible(true);
      setNotification(state.notification);
    }
  }, [state.notification]);
  return (
    <>
      {notification && (
        <Snackbar
          visible={isVisible}
          onDismiss={() => setVisible(false)}
          duration={notification.duration ? notification.duration : 5000}
          action={{
            icon: notification.icon ? notification.icon : "",
            label: notification.buttonLabel ? notification.buttonLabel : "",
            onPress: () => {},
          }}
        >
          {notification.message}
        </Snackbar>
      )}
    </>
  );
};
