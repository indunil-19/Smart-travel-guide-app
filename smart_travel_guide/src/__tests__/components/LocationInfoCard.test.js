import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import { LocationInfoCard } from "../../components/LocationInfoCard";

test("should match snapshot", () => {
  const tree = renderer
    .create(
      <LocationInfoCard
        photo={"imglink"}
        location={{
          name: "Galle Face Beach",
          formatted_address: "Galle Face,Colombo",
          icon: "iconLink",
          rating: 4,
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
