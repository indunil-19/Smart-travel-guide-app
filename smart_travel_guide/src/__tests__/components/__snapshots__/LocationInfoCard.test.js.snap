// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "rgba(0, 0, 0, 0.12)",
      "borderRadius": 10,
      "elevation": 6,
      "marginVertical": 12,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 6,
      "width": "90%",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#eeeeee",
            "height": 195,
            "overflow": "hidden",
          },
          undefined,
          Object {},
        ]
      }
    >
      <Image
        siblings={
          Array [
            "withTheme(Card.Cover)",
            "withTheme(Card.Title)",
            "Card.Content",
          ]
        }
        source={1}
        style={
          Array [
            Object {
              "flex": 1,
              "height": undefined,
              "justifyContent": "flex-end",
              "padding": 16,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
          },
          Object {
            "minHeight": 72,
          },
          Object {
            "paddingHorizontal": 10,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  Array [
                    Object {
                      "minHeight": 30,
                    },
                    Object {
                      "marginBottom": 0,
                    },
                    undefined,
                  ],
                ],
              ],
            ]
          }
        >
          Galle Face Beach
        </Text>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 12,
                    "letterSpacing": 0.4,
                    "lineHeight": 20,
                    "marginVertical": 2,
                  },
                  Array [
                    Object {
                      "marginVertical": 0,
                      "minHeight": 20,
                    },
                    undefined,
                  ],
                ],
              ],
            ]
          }
        >
          Galle Face,Colombo
        </Text>
      </View>
      <View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={false}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 11.25,
                  "height": 22.5,
                  "width": 22.5,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          <View>
            <Image
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              source={
                Object {
                  "uri": "iconLink",
                }
              }
              style={
                Array [
                  Object {
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "height": 15,
                    "resizeMode": "contain",
                    "tintColor": "#2DD4BF",
                    "width": 15,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
          },
          Object {
            "paddingBottom": 16,
          },
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingRight": 5,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "column",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 20,
                  "margin": 3,
                  "tintColor": "#2DD4BF",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                  "width": 20,
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 20,
                  "margin": 3,
                  "tintColor": "#2DD4BF",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                  "width": 20,
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 20,
                  "margin": 3,
                  "tintColor": "#2DD4BF",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                  "width": 20,
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 20,
                  "margin": 3,
                  "tintColor": "#2DD4BF",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                  "width": 20,
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 20,
                  "margin": 3,
                  "tintColor": "#BDC3C7",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                  "width": 20,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Learn More
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
